File: src/webui/app/logic/domain/time-plan-activity-kind.ts
import { TimePlanActivityKind } from "@jupiter/webapi-client";

export function timePlanActivityKindName(kind: TimePlanActivityKind): string {
  switch (kind) {
    case TimePlanActivityKind.FINISH:
      return "Fisish";
    case TimePlanActivityKind.MAKE_PROGRESS:
      return "Make Progress";
  }
}

const TIME_PLAN_ACTIVITY_KIND_MAP = {
  [TimePlanActivityKind.FINISH]: 0,
  [TimePlanActivityKind.MAKE_PROGRESS]: 1,
};

export function compareTimePlanActivityKind(
  kind1: TimePlanActivityKind,
  kind2: TimePlanActivityKind
): number {
  return (
    TIME_PLAN_ACTIVITY_KIND_MAP[kind1] - TIME_PLAN_ACTIVITY_KIND_MAP[kind2]
  );
}

File: src/core/jupiter/core/domain/concept/time_plans/time_plan_activity_kind.py
"""The kind of an activity."""

from functools import total_ordering

from jupiter.core.framework.value import EnumValue, enum_value


@total_ordering
@enum_value
class TimePlanActivityKind(EnumValue):
    """The kind of a time plan activity."""

    FINISH = "finish"
    MAKE_PROGRESS = "make-progress"

    def __lt__(self, other: object) -> bool:
        """Compare this with another."""
        if not isinstance(other, TimePlanActivityKind):
            raise Exception(
                f"Cannot compare with {other.__class__.__name__}",
            )

        all_values = self.get_all_values()

        return all_values.index(self.value) < all_values.index(other.value)
